docs/enhancements/consumers-scheduling.md

```diff
@@ -0,0 +1,111 @@
+# Consumers Scheduling
+
+## Goals
+
+- Improve Scalability
+  - Remove limit on the number of triggers per broker - https://github.com/knative-sandbox/eventing-kafka-broker/issues/337
+  - Consumers partitioning and Trigger parallelism https://github.com/knative-sandbox/eventing-kafka-broker/issues/785
+- Give to the event consumers control on the number of replicas (delivery parallelism)
+- Minimal changes to the data plane
+- Ease the implementation of a new MT `KafkaSource`
+
+## Introduction
+
+The `Trigger` and `KafkaSource` controller will schedule resources by using the `Placeable` duck type.
+
+```go
+// Placeable is a list of podName and virtual replicas pairs.
+// Each pair represents the assignment of virtual replicas to a pod
+type Placeable struct {
+	MaxAllowedVReplicas *int32      `json:"maxAllowedVReplicas,omitempty"`
+	Placement           []Placement `json:"placements,omitempty"`
+}
+
+type Placement struct {
+	// PodName is the name of the pod where the resource is placed
+	PodName string `json:"podName,omitempty"`
+	// VReplicas is the number of virtual replicas assigned to in the pod
+	VReplicas int32 `json:"vreplicas,omitempty"`
+}
+```
+
+The `KafkaSource` type embeds the `Placeable` type in the status, while the Trigger will use
+a different representation of that type in the `status.annotations` field like the following:
```

Comment:
will the Trigger then be implementing the duck type? i.e any code dealing with `Placeable` can still deal with the Triggers?

Reply:
/cc @aavarghese

Reply:
@devguyio since Placements info will be added as Annotations for Trigger (and could do the same for KafkaSource, as well), don't think Trigger will implement this duck type. Pod reconcilers dealing with these placements use them independent of resource being a Trigger or Kafkasource. (Apologize if I misunderstood what I'm learning about the scheduling proposal)